file.go

```diff
@@ -701,10 +704,10 @@ var (
 	bufioReaderPool sync.Pool
 )
 
-func getBufioReader(r io.Reader) *bufio.Reader {
+func getBufioReader(r io.Reader, bufferSize int) *bufio.Reader {
```

Comment:
An issue with this approach is that we end up with a single pool of buffers with different sizes. When the pool fills up, the request buffer size ends up not being respected (unless the application uses a single buffer size).

An option could be to maintain multiple pools for different buffer sizes (e.g. multiple of 4096, up to a sane limit). Then we round up the requested buffer size to the nearest multiple to offer a better guarantee to the application that we will be respecting the requested size.

What do you think?

Reply:
It did not occur to me that this was a shared pool that all parquet.File's access. Good catch. I consider it reasonable to assume that most applications would have at most 2 or 3 different buffer sizes throughout. We could maintain a `map[buffer_size] -> sync.Pool` ?

Reply:
Sounds good 👍